client/txn.go

```diff
@@ -422,9 +417,16 @@ func (txn *Txn) send(calls ...proto.Call) error {
 		return err
 	}
 
-	if haveEndTxn {
-		// We sent an EndTransaction and it succeeded.
-		txn.finishedOK = true
+	// We sent an EndTransaction and it succeeded. Accept the Transaction
+	// returned from the server.
+	if haveEndTxn && len(calls) > lastIndex {
+		// TestCommitTransactionOnce needs this null check. Why?
```

Comment:
it probably sends a hand-made response which isn't properly initialized. The normal path updating the Txn uses `.Update()`, which is more robust than what you're doing in 425. Not an actual issue now that you're removing that part.